### PR TITLE
Add manifest.json and PWA support

### PR DESCRIPTION
**Pull request by cairocoder01**
This provides support for install to home screen on browsers that support it.

There is some placeholder js code to override the install prompt, but I left that turned off for now so that only the default browser prompts will happen now. I don't think most browsers actually prompt, but Chrome adds an icon in the address bar to manually install a PWA. Until we do a front-end rework to clean things up more, that's probably sufficient.

This also adds a framework for looking into other PWA features such as push notifications or offline access.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -4,6 +4,10 @@
         "wp-coding-standards/wpcs": "3.1.*",
         "squizlabs/php_codesniffer": "*"
     },
+    "scripts": {
```
Comment (corsacca) [inline composer.json:7]: @cairocoder01 how are these scripts used?

**Comment by corsacca** [comment]
This will be cool!

File: composer.json (hunk 1 above)
Comment (cairocoder01) [reply composer.json:7]: composer run lint

File: dt-core/configuration/config-site-defaults.php [hunk 2]
```diff
@@ -675,7 +675,7 @@ function dt_header_icon_and_meta(){
     <link rel="apple-touch-icon" sizes="180x180" href="<?php echo esc_url( get_template_directory_uri() ); ?>/dt-assets/favicons/apple-touch-icon.png">
     <link rel="icon" type="image/png" sizes="32x32" href="<?php echo esc_url( get_template_directory_uri() ); ?>/dt-assets/favicons/favicon-32x32.png">
     <link rel="icon" type="image/png" sizes="16x16" href="<?php echo esc_url( get_template_directory_uri() ); ?>/dt-assets/favicons/favicon-16x16.png">
-    <link rel="manifest" href="<?php echo esc_url( get_template_directory_uri() ); ?>/dt-assets/favicons/site.webmanifest">
+    <link rel="manifest" href="<?php echo esc_url( home_url( '/manifest.json' ) ); ?>">
```
[the remark this answers is not in the document]
Comment (cairocoder01) [reply dt-core/configuration/config-site-defaults.php:678]: @corsacca Yes, it is. The other code in this should create that URL and respond with the appropriate json content.

[the remark this answers is not in the document]
Comment (corsacca) [reply dt-core/configuration/config-site-defaults.php:678]: Awesome. It probably was on the wrong branch when i tested.

**Pull request merged**